File: RadVolontera/RadVolontera.Services/Database/VolunteeringAnnouncement.cs
using RadVolontera.Services.Domain.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RadVolontera.Services.Database
{
    public class VolunteeringAnnouncement : BaseSoftDeleteEntity
    {
        public long Id { get; set; }
        public string Place { get; set; }
        public long AnnouncementStatusId { get; set; }
        public  Status AnnouncementStatus { get; set; }
        public long CityId { get; set; }
        public City City { get; set; }
        public string MentorId { get; set; }
        public User Mentor { get; set; }
        public int TimeFrom { get; set; }
        public int TimeTo { get; set; }
        public DateTime Date { get; set; }
        public string Notes { get; set; }
        public  Report Report { get; set; }
    }
}

File: RadVolontera/UI/radvolontera_mobile/lib/models/volunteering_announcement/volunteering_announcement.dart



import 'package:json_annotation/json_annotation.dart';

import '../account/account.dart';
import '../city/city.dart';
import '../status/status.dart';

part 'volunteering_announcement.g.dart';
@JsonSerializable()
class VolunteeringAnnouncementModel {
  int? id;
  String? place;
 int? announcementStatusId;
  StatusModel? announcementStatus;
  int? cityId;
  CityModel? city;
  String? mentorId;
  AccountModel? mentor;
  String? timeFrom;
  String? timeTo;
 String? notes;
 DateTime ? date;
 bool hasReport;
 String ? reason;
VolunteeringAnnouncementModel(this.id, this.place, this.announcementStatusId, 
this.announcementStatus,this.cityId,this.city,this.mentorId,this.mentor,this.hasReport,this.reason);
      /// A necessary factory constructor for creating a new User instance
  /// from a map. Pass the map to the generated `_$UserFromJson()` constructor.
  /// The constructor is named after the source class, in this case, User.
   factory VolunteeringAnnouncementModel.fromJson(Map<String, dynamic> json) => _$VolunteeringAnnouncementModelFromJson(json);

  // /// `toJson` is the convention for a class to declare support for serialization
  // /// to JSON. The implementation simply calls the private, generated
  // /// helper method `_$UserToJson`.
   Map<String, dynamic> toJson() => _$VolunteeringAnnouncementModelToJson(this);
}